src/main/java/net/neoforged/moddevgradle/dsl/NeoForgeExtension.java

```diff
@@ -1,177 +1,50 @@
-     * You can override this property to use i.e. MCP for up to 1.20.1.
-     * <p>
-     * This property is mutually exclusive with {@link #getNeoForgeArtifact()}.
-     *
-     * @return Maven artifact coordinate (group:module:version)
-     */
-    public abstract Property<String> getNeoFormArtifact();
-
-    /**
-     * The list of additional access transformers that should be applied to the Minecraft source code.
-     * <p>
-     * If you do not set this property, the plugin will look for an access transformer file at
-     * {@code META-INF/accesstransformer.cfg} relative to your main source sets resource directories.
-     *
-     * @see <a href="https://projects.neoforged.net/neoforged/accesstransformers">Access Transformer File Format</a>
-     */
-    public void accessTransformers(Action<DataFileCollection> action) {
-        action.execute(accessTransformers);
-    }
-
-    public DataFileCollection getAccessTransformers() {
-        return accessTransformers;
+    @Deprecated(forRemoval = true)
+    public void setVersion(Object any) {
+        throw new InvalidUserCodeException("Please use enableModding { neoForgeVersion = ... } instead of the version property.");
     }
 
-    /**
-     * Replaces current access transformers.
-     */
-    public void setAccessTransformers(Object... paths) {
-        getAccessTransformers().getFiles().setFrom(paths);
-    }
-
-    /**
-     * The data-files describing additional interface implementation declarations to be added to
-     * Minecraft classes.
-     * <p>
-     * <strong>This is an advanced property: Injecting interfaces in your development environment using this property will not implement
-     * the interfaces in your published mod. You have to use Mixin or ASM to do that.</strong>
-     *
-     * @see <a href="https://github.com/neoforged/JavaSourceTransformer?tab=readme-ov-file#interface-injection">Interface Injection Data Format</a>
-     */
-    public void interfaceInjectionData(Action<DataFileCollection> action) {
-        action.execute(interfaceInjectionData);
-    }
-
-    public DataFileCollection getInterfaceInjectionData() {
-        return interfaceInjectionData;
-    }
-
-    /**
-     * Replaces current interface injection data files.
-     */
-    public void setInterfaceInjectionData(Object... paths) {
-        getInterfaceInjectionData().getFiles().setFrom(paths);
-    }
+    public void enableModding(Action<ModdingVersionSettings> customizer) {
+        var modDevPlugin = project.getPlugins().getPlugin(ModDevPlugin.class);
 
-    /**
-     * Enable access transformer validation, raising fatal errors if an AT targets a member that doesn't exist.
-     * <p>
-     * <b>Default</b> {@code false}<br>
-     */
-    public abstract Property<Boolean> getValidateAccessTransformers();
+        var settings = project.getObjects().newInstance(ModdingVersionSettings.class);
+        // By default, enable modding deps only for the main source set
```

Comment:
I'm not sure I agree with this being *just a convention*. By far the most common use case is to add and not overwrite, so `enabledSourceSets += sourceSets.api` should be fine.